### PR TITLE
redux-header

### PR DESCRIPTION
**Pull request by kcbangi**
setup redux, header, and made some changes on the list of habits and move it to reducers.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -4,7 +4,30 @@ import './index.css';
 import App from './App/App';
 import * as serviceWorker from './serviceWorker';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+//redux
+import { createStore } from 'redux';
+import { Provider } from "react-redux";
+import reducers from './reducers/'
+
+// import { createStore, applyMiddleware, compose } from 'redux';
```
Comment (danieltichiyama) [inline src/index.js]: it looks like you set it up to not use the applyMiddleware or redux-thunk module.  Can you explain why you decided to do it that way?

File: src/reducers/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { LOAD_HABIT } from '../actions';
```
Comment (danieltichiyama) [inline src/reducers/index.js]: change to LOAD_HABITS to keep the same as the function's name.

File: src/reducers/index.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      name: 'Not taking shower',
+      category: 'Hygiene'
+    }
+  ]
+}
+
+let reducer = (state = defaultState, action) => {
+  switch (action.type) {
+    case LOAD_HABIT:
+      return state.books;
```
Comment (danieltichiyama) [inline src/reducers/index.js]: state.books doesn't exist.  and I'm pretty sure all of these cases in the switch statement need to return an object.  The object returned becomes the new state, so it'd have to include all of the old state things, and whatever changes your function made.  I believe that's why we need to use Object.assign(), to combine the old and new state objects.

File: src/actions/index.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+//action types
+export const LOAD_HABIT = 'LOAD_HABIT';
```
Comment (danieltichiyama) [inline src/actions/index.js]: change the variable to LOAD_HABITS and the value to "LOAD_HABITS" to keep it the same name as the function loadHabits().

File: src/actions/index.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+//action create
+export function loadHabits(payload) {
```
Comment (danieltichiyama) [inline src/actions/index.js]: you don't need the payload parameter if there is no parameter being passed into it from the component.

File: src/actions/index.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+  return {
+    type: LOAD_HABIT,
+    payload
```
Comment (danieltichiyama) [inline src/actions/index.js]: set payload to equal null

**Review by danieltichiyama — CHANGES_REQUESTED** [review]
Can you please provide details on how your changes should work and, if possible, instructions for how to test them?

File: src/App/App.js [hunk 7]
```diff
@@ -1,15 +1,46 @@
-import React from "react";
+import React, { Component } from "react";
+import { connect } from 'react-redux';
+import { loadHabits } from '../actions';
 import "./App.module.css";
-import HabitList from "../containers/HabitList/HabitList";
+
+import Header from '../components/Header'
+import HabitList from "../containers/HabitList";
 import NavigationComponent from "../containers/NavigationComponent.jsx/NavigationComponent";
```
Comment (jkhirano) [inline src/App/App.js]: This threw an error. It should be revised to:
```
import NavigationComponent from "../containers/NavigationComponent";
```

**Review by danieltichiyama — APPROVED** [review]
please check with Janelle's comment.

**Review by jkhirano — APPROVED** [review]
Just need to fix the App.js navigation component and then you can merge

**Pull request merged**